.azure-pipelines/ultimate-pipeline.yml

```diff
@@ -610,6 +635,8 @@ stages:
 
       - script: tracer\build.cmd BuildAndRunManagedUnitTests --code-coverage
         displayName: Build and Test
+        env:
+          DD_LOGGER_DD_API_KEY: $(ddApiKey)
```

Comment:
I assume this isn't actually needed for the windows stages, as we already set it "globally" in `variables`. I'm not _entirely_ convinced it's necessary to pass as an explicit param for the `run-in-docker.sh` template, but not going to fight you on that one 😄  I've added suggestions for all the ones I _think_ we can remove, but feel free to ignore/reject if I'm wrong 🙂 

```suggestion
```

Reply:
secrets cannot be set `globally` (doesn't map to env-vars) so we have to do it in each job.